### PR TITLE
DAOS-2172 doc: update DAOS service section

### PR DESCRIPTION
**Pull request by johannlombardi**
- add new diagrams for components, layering & services to src/README.md
- move all media files to doc/graph
- populate generic section introducing the different DAOS services
  and infrastructure libraries.

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
Rendered here: https://github.com/johannlombardi/daos/blob/johann/doc/src/README.md

File: src/README.md [hunk 1]
```diff
@@ -89,31 +92,61 @@ Like gRPC, RPC are serialized via protocol buffers.
...
+
+![DAOS Internal Services & Libraries](/doc/graph/services.png)
+
+A DAOS service can be spread across the control and data planes and communicate internally through dRPC. Most services have a client and server components that can synchronize through gRPC or CART. Cross-service communications are always done through direct API calls. Those function calls can be invoked across either the client or server component of the services. While each DAOS service is designed to be fairly autonomous and isolated, some are more tightly coupled than others. That is typically the case of the rebuild service that needs to interact closely with the pool, container and object services to restore data redundancy after a DAOS server failure.
+
+While the service-based architecture offers flexibity and extensibility, it is combined with a set of infrastucture libraries that provide a rich software ecosystem (e.g. communications, persistent storage access, asynchronous task execution with dependency graph, accelerator support, ...) accessible to all the DAOS services.
```
Comment (kjacque) [inline src/README.md]: spelling - "flexibility"

File: src/README.md [hunk 2]
```diff
@@ -89,31 +92,61 @@ Like gRPC, RPC are serialized via protocol buffers.
...
 
 <a id="31"></a>
 ### Architecture
 
-The DAOS stack is organized as a set of microservices over a client/server architecture. The <a href="#5a">figure</a> below shows the logical layering of the DAOS stack.
+As shown in the diagram below, the DAOS stack is structured as a collection of storage services over a client/server architecture. Examples of DAOS services are the pool, container, object and rebuild services.
+
+![DAOS Internal Services & Libraries](/doc/graph/services.png)
+
+A DAOS service can be spread across the control and data planes and communicate internally through dRPC. Most services have a client and server components that can synchronize through gRPC or CART. Cross-service communications are always done through direct API calls. Those function calls can be invoked across either the client or server component of the services. While each DAOS service is designed to be fairly autonomous and isolated, some are more tightly coupled than others. That is typically the case of the rebuild service that needs to interact closely with the pool, container and object services to restore data redundancy after a DAOS server failure.
```
Comment (kjacque) [inline src/README.md]: spelling - "Most services have a client and server components" -> "Most services have client and server components"

File: src/README.md [hunk 3]
```diff
@@ -89,31 +92,61 @@ Like gRPC, RPC are serialized via protocol buffers.
...
+- <a href="bio/README.md">Blob I/O (BIO)</a>
+- <a href="placement/README.md">Algorithmic object placement</a>
+- <a href="rdb/README.md">Replicated database (RDB)</a>
+- <a href="rsvc/README.md">Replicated service framework (RSVC)</a>
+
+<a id="34"></a>
+### DAOS Services
+
+The diagram below shows the internal layering of the DAOS services and interactions with the different libraries mentioned above.
+![DAOS Internal Layering](/doc/graph/layering.png)
```
Comment (kjacque) [inline src/README.md]: Looks like right now the diagram only shows dRPC communications going from the data plane to control plane for security. The flow for manipulating ACLs is: dmg -> control plane (via gRPC) -> data plane (via dRPC)

Comment (johannlombardi) [reply src/README.md]: right, that was for the signature verification. Will fix that.

**Comment by johannlombardi** [comment]
Sorry for the forced push ...

**Pull request merged**